### ISSUE TITLE
BUG: Benchmark repetitions reuse the first split's cached model

### ISSUE THREAD

**Issue by Nikhi00718** [shap/shap#5119]
### Problem

`shap.benchmark.metrics.__score_method` averages a metric over several train/test splits, but the on-disk model cache key does not include the repetition/split index:

```python
model_id = "model_cache__v" + "__".join([__version__, data_hash, model_generator.__name__]) + ".pickle"
```

On the first repetition the model is fitted to `X_train` for split 0 and cached. Every later repetition finds the same file and reloads that first model even though `train_test_split(..., random_state=i)` produced a different `X_train` and `X_test`.

The attribution cache already includes `i`, so model and attribution caching currently disagree: each repetition computes an attribution for its own split using a model trained on split 0.

### Expected behavior

Each benchmark repetition should fit/cache a model for its own train/test split. Including `i` in the model cache key preserves reuse across benchmark runs while preventing reuse across different splits.

### Proposed validation

A focused regression test can run two repetitions with a counting model generator and assert that two independently fitted models are created and two model-cache files exist.

### Duplicate search

I searched existing issues and pull requests for the benchmark model cache key and split reuse and found no matching report. The open NumPy RNG-state issues concern a separate problem in the same legacy benchmark module.

### AI assistance disclosure

Codex autonomously audited the benchmark implementation, identified and de-duplicated this defect, and drafted this report. A minimal disclosed draft PR with a regression test will follow.
